lib/ansible/modules/net_tools/nsupdate.py

```diff
@@ -326,12 +331,26 @@ def record_exists(self):
             response = self.__do_update(update)
             self.dns_rc = dns.message.Message.rcode(response)
             if self.dns_rc == 0:
-                return 1
+                if self.ttl_changed():
+                    return 2
+                else:
+                    return 1
             else:
                 return 2
         else:
             return 0
 
+    def ttl_changed(self):
+        query = dns.message.make_query(self.fqdn, self.module.params['type'])
+
+        try:
+            lookup = dns.query.tcp(query, self.module.params['server'], timeout=10, port=self.module.params['port'])
+        except (socket_error, dns.exception.Timeout) as e:
+            self.module.fail_json(msg='DNS server error: (%s): %s' % (e.__class__.__name__, to_native(e)))
+
+        current_ttl = lookup.answer[0].ttl
```

Comment:
what if empty result ? This should be tested
also we can have a different TTL per record value, not only per record if i'm correct

Reply:
In which scenario would we get an empty result? Wouldn't we have taken a different code path in that scenario?

A different TTL per value? How would that even work DNS wise?

Reply:
I just tested that in my bind

`$ nsupdate`
```
> server localhost
> update add ttldemo.msg.lol. 300 TXT "TTL 300"
> update add ttldemo.msg.lol. 600 TXT "TTL 600"
> send
```
After syncing the changes to disk both entries in the zone file have a TTL of 600.

`$ dig txt ttldemo.msg.lol @127.0.0.1`
```
;; ANSWER SECTION:
ttldemo.msg.lol.	600	IN	TXT	"TTL 300"
ttldemo.msg.lol.	600	IN	TXT	"TTL 600"
```
I haven't checked RFCs but at least bind doesn't seem to support that.

Reply:
https://tools.ietf.org/html/rfc2181#section-5.2

> ...
> Consequently the use of differing TTLs in an RRSet is hereby deprecated, the TTLs of all RRs in an RRSet must be the same.
> ...

Reply:
okay thanks for the clarification.